src/main/external-resources/nsis/pms.nsi

```diff
@@ -117,16 +117,14 @@ Function GetJRE
 		Call CheckJREVersion
```

Comment:
@SubJunk I'm really confused by the whole ```GetJRE``` function. First of all, the syntax isn't entirely clear to me and pushing and popping (I assume) to/from the stack is confusing me. Since you have C experience I guess some of this makes more sense to you.

Assuming that the above haven't thrown me completely off here, it seems to me that there are some problems here: 
* I can't see that steps 1, 2 or 4 is performed. Step 1. would be futile, but shouldn't it then be removed from the description? The other steps would be useful in case Java isn't installed with Oracle's installer.
* Some registry locations seems to be checked multiple times, while the use of ```SetRegView``` 32/64 and ```Wow6432Node``` really confuse me. As far as I understand the ```Wow6432Node``` is the "32 bit view" on a 64 bit OS, and shouldn't need to be access directly. Setting the "reg view" to 32 should give the same result without explicitly specifying ```Wow6432Node```. This seems to be a combination of both strategies, with redundant checks. Your Java 9 check will, as far as I can tell, fail to detect a 32 bit Java 9 installed on a 64 bit OS. What I think needs to be checked is:
  * On a 64 bit OS both ```HKLM "SOFTWARE\JavaSoft\Java Runtime Environment"``` and ```HKLM "SOFTWARE\JavaSoft\JRE"``` in both 32 and 64 "registry view" - a total of 4 checks.
  * On a 32 bit OS both ```HKLM "SOFTWARE\JavaSoft\Java Runtime Environment"``` and ```HKLM "SOFTWARE\JavaSoft\JRE"```. Since "registry view" isn't a concept for 32 bit OS'es, that means a total of 2 checks.